### ISSUE TITLE
Unsaved Changes dialog not working correctly in Timelines

### ISSUE THREAD

**Issue by charlesh88** [nasa/openmctweb#660]
Testathon 20160209

When a user is editing in NEM and clicks the "X" to cancel editing in the upper right of the interface, the unsaved changes dialog is displayed. If the user clicks "Cancel" in that dialog, they should be taken back to edit mode. Instead, edit mode is actually exited *before* the dialog is displayed, and the system is put into a weird half edit-mode, half browse-mode situation: Save/Cancel buttons are not displayed, the Create button *is* displayed, but clicking anywhere in the tree results in the unsaved changes dialog being displayed as if the user was still editing.

Ideally, clicking the cancel edit "X" button should have the same result as if the user tried to navigate elsewhere while editing: the unsaved changes dialog is displayed and clicking Cancel in that dialog should keep the user in edit mode.

**Comment by akhenry** [comment]
This is addressed in https://github.com/nasa/openmctweb/issues/627 Since it's critical, I could try and address it separately before Wednesday if need be.

**Comment by akhenry** [comment]
> Ideally, clicking the cancel edit "X" button should have the same result as if the user tried to navigate elsewhere while editing: the unsaved changes dialog is displayed and clicking Cancel in that dialog should keep the user in edit mode.

This has been broken out into a separate issue - https://github.com/nasa/openmctweb/issues/664

**Comment by akhenry** [comment]
Have confirmed after recreating successfully in WARP and master that it is indeed already fixed in #627

**Comment by VWoeltjen** [comment]
Is it possible to separate out the changes which address (or partially address) this from #651? Would like to address this in isolation before introducing new functionality (plot options)

**Comment by akhenry** [comment]
The bits of #627 that address this do not directly depend on #651. Extricating just those bits from the branch will inevitably involve some duplication of effort though, as they'll have to be shoehorned into the browse controller and markup in master, changes which will then be discarded when #627 is merged. It could totally be done in the timeframe though.

**Comment by VWoeltjen** [comment]
Got it. Don't want to waste too much effort, but also see that #651 is still undergoing changes for review and don't want to rush that.

Given that, I'd propose to base the decision on a ballpark time estimate for cherry-picking (in  the git sense and/or manual copy+paste sense). 

* If we're talking hours or days (not sure how deeply those changes are embedded in other browse mode changes), then I think we should focus on getting #651 and #627 merged; this is consistent with the spirit of code freeze (it's the most efficient solution to a blocker.) 
* If it's shorter than that, then the duplicated effort is pretty minimal, and I think the procedural risk mitigation would make it worthwhile to separate out just the changes which address this issue.

@charlesh88 - what are your thoughts on these approaches?

**Comment by akhenry** [comment]
Have created a PR #665 for a fix for this. Turns out it's fairly straightforward to fix in master. The code in error was introduced as part of NEM, and I have applied a fix to this code. Might be possible to refactor out the NEM code here as part of #627

**Comment by RCarvalhoNASA** [comment]
Testathon 20160211 I went in to edit a timeline, added an activity, then started to attempt to test this (close without save by clicking the X).  It simply let me do so without any kind of warning dialog that I would lose my changes.

**Comment by akhenry** [comment]
@RCarvalhoNASA Unfortunately a known issue for this sprint, there is a ticket to address this - #664

**Comment by akhenry** [comment]
Testathon 20160211

Verified as fixed.
